### PR TITLE
CppCheck Data_files and other clean up

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes CppCheck style suggestions
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by Myoldmopar** [comment]
Actually there is a conflict here, so I'll exclude this from my lumped testing.  It looks to be a tiny conflict FWIW.

File: src/EnergyPlus/DataZoneEquipment.cc [hunk 1]
```diff
@@ -757,12 +743,12 @@ void GetZoneEquipmentData(EnergyPlusData &state)
     // Allocate TermUnitSizing array and set zone number
     if (locTermUnitSizingCounter > 0) {
         state.dataSize->NumAirTerminalUnits = locTermUnitSizingCounter;
-        TermUnitSizing.allocate(state.dataSize->NumAirTerminalUnits);
+        state.dataSize->TermUnitSizing.allocate(state.dataSize->NumAirTerminalUnits);
         for (int loopZoneNum = 1; loopZoneNum <= state.dataGlobal->NumOfZones; ++loopZoneNum) {
             {
-                auto &thisZoneEqConfig(state.dataZoneEquip->ZoneEquipConfig(loopZoneNum));
+                auto &thisZoneEqConfig = state.dataZoneEquip->ZoneEquipConfig(loopZoneNum);
```
Comment (rraustad) [inline src/EnergyPlus/DataZoneEquipment.cc:749]: @amirroth I was changing this notation and noticed your comment this morning, constructor notation for scalars only. Revert change?

Comment (amirroth) [reply src/EnergyPlus/DataZoneEquipment.cc:749]: 1. A reference is a scalar.
2. I made a similar comment about references later.
3. Leave it, thx!!

**Comment by Myoldmopar** [comment]
This one is super duper green on CI, testing it next.

**Comment by Myoldmopar** [comment]
Pulled develop from just now, built fine, tested fine, no regressions, merging this one.  Thanks for another one @rraustad

**Pull request merged**